osu.Game/OsuGame.cs

```diff
@@ -445,10 +445,17 @@ public void HandleLink(LinkDetails link) => Schedule(() =>
                     break;
 
                 case LinkAction.SearchBeatmapSet:
-                    if (link.Argument is RomanisableString romanisable)
-                        SearchBeatmapSet(romanisable.GetPreferred(Localisation.CurrentParameters.Value.PreferOriginalScript));
-                    else
-                        SearchBeatmapSet(argString);
+                    SearchBeatmapSet(argString);
+                    break;
+
+                case LinkAction.SearchBeatmapTitle:
+                    string title = ((RomanisableString)link.Argument).GetPreferred(Localisation.CurrentParameters.Value.PreferOriginalScript);
```

Comment:
Any reason for direct cast rather than supporting `string`s as well? I'd say for future-proofing you should follow the original patter of type matching.